sphinx/writers/latex.py

```diff
@@ -400,15 +400,26 @@ def get_colspec(self):
             total = sum(self.colwidths)
             colspecs = ['\\X{%d}{%d}' % (width, total) for width in self.colwidths]
             return '{|%s|}\n' % '|'.join(colspecs)
-        elif self.has_problematic:
-            return '{|*{%d}{\\X{1}{%d}|}}\n' % (self.colcount, self.colcount)
-        elif self.get_table_type() == 'tabulary':
-            # sphinx.sty sets T to be J by default.
-            return '{|' + ('T|' * self.colcount) + '}\n'
-        elif self.has_oldproblematic:
-            return '{|*{%d}{\\X{1}{%d}|}}\n' % (self.colcount, self.colcount)
+        elif not self.get_table_type() == 'tabulary':
+            total_Y, colcount_l = 0.0, 0
+            for i in range(self.colcount):
+                estimate = self.empirical_colwidths[i]
+                if estimate > 16:
```

Comment:
this threshold at 16 was chosen also because any kind of LaTeX environment will force use of a ``\Y`` type, not an ``l`` column.

The idea is that narrow cells should be rendered in l columns. Cells with more than a few words should use p type columns with a width proportional to contents.

This is what LaTeX macro package tabulary does, but it looks easier on Python side, although we want to measure only the weight of rst source (is thaat ok with substitutions?)

Reply:
+1 basically; but we have to improve this a little.

* add comment for description
* replace 16 with constant (it is magic number!)